html_app/build.py

```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/env python
+
+"""The static asset pipeline for StarCellBio.
```

Comment:
Accepted arguments are always strings, and I don't think there's a way to validate that the argument passed was an integer. The example on http://docopt.org/ shows choices.

Here's the docstring from that silly deckgen script as an example including nargs and type implication:
```manual
Standalone slide deck collection builder.

Usage:
  deckgen.py build [-s SRC] [-d DEST] [-t TEMP] [NAME...]
  deckgen.py watch [-s SRC] [-d DEST] [-t TEMP]
  deckgen.py -h | --help

Arguments:
  name   The name of a slide deck to generate. Accompanying markdown should be
         located in the source directory. For example, if the markdown file is
         named "slides.md", the name "slides" should be passed.
         If no names are passed, the default behavior is to run against ALL
         markdown files in the source directory

Options:
  -h --help               Show this help screen.
  -s --source=SRC         Source markdown directory [default: markdown].
  -d --destination=DEST   Destination HTML directory [default: decks].
  -t --template=TEMP      HTML template to use [default: template.html].
```

That being said, docopt is indeed waaaay too magical and doesn't scale for really complicated CLIs and situations requiring a lot of fine grained control, but I figured it would be okay here.